File: pid-backend/src/main/java/cz/fit/cvut/pidbackend/Model/Dto/TripResponse.java
package cz.fit.cvut.pidbackend.Model.Dto;

import cz.fit.cvut.pidbackend.Model.Shape;
import cz.fit.cvut.pidbackend.Model.Vehicle;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;
import java.util.Set;

@Data
@NoArgsConstructor
@AllArgsConstructor(access = AccessLevel.PUBLIC)
public class TripResponse {
    private TripDto trip;
    private Vehicle vehicle;
    private Set<Shape> shapes;
    private List<DelayDto> delays;
}

File: android-app/app/src/main/java/cvut/fit/pidmobapp/model/response/TripResponse.kt
package cvut.fit.pidmobapp.model.response

import cvut.fit.pidmobapp.model.Trip
import cvut.fit.pidmobapp.model.Shape
import cvut.fit.pidmobapp.model.Vehicle
import cvut.fit.pidmobapp.model.dto.DelayDto

data class TripResponse(
    val trip: Trip,
    val vehicle: Vehicle,
    val shapes: List<Shape>?,
    val delays: List<DelayDto>
)
